beekeeper-cleanup/src/main/java/com/expediagroup/beekeeper/cleanup/path/aws/S3PathCleaner.java

```diff
@@ -53,30 +56,22 @@ public void cleanupPath(HousekeepingPath housekeepingPath) {
     S3SchemeURI s3SchemeURI = extractURI(housekeepingPath.getPath());
     String key = s3SchemeURI.getKey();
     String bucket = s3SchemeURI.getBucket();
-
-    // doesObjectExists returns true only when the key is a file
+    S3BytesDeletedCalculator bytesDeletedCalculator = new S3BytesDeletedCalculator(s3Client);
     boolean isFile = s3Client.doesObjectExist(bucket, key);
-
-    if (isFile) {
-      long size = s3Client.getObjectMetadata(bucket, key).getContentLength();
-      s3Client.deleteObject(bucket, key);
-      s3BytesDeletedReporter.reportSize(size);
-    } else {
-      deleteFilesInDirectory(bucket, key);
-
-      try {
-        String path = s3SchemeURI.getPath();
-        if (path.endsWith("/")) {
-          path = path.substring(0, path.length() - 1);
-        }
-        sentinelFilesCleaner.deleteSentinelFiles(path);
-
-        // attempt to delete parents if there is at least one parent
-        if (key.contains("/")) {
-          deleteParentSentinelFiles(bucket, key, path, housekeepingPath.getTableName());
-        }
-      } catch (Exception e) {
-        log.warn("Sentinel file(s) could not be deleted", e);
+    try {
+      if (isFile) {
+        deleteFile(bucket, key, bytesDeletedCalculator);
+      } else {
+        deleteFilesInDirectory(bucket, key, bytesDeletedCalculator);
+        deleteSentinelFiles(s3SchemeURI, key, bucket, housekeepingPath.getTableName());
+      }
+    } catch (Exception e) {
```

Comment:
Is this catch block necessary? Isn't the default behaviour to do what it's doing?

Reply:
Yes true, will remove the catch block